bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,216 @@
+public class SecureHttpClientFactory implements HttpClientFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(SecureHttpClientFactory.class);
+
+    private static final String CONFIG_MIN_THREADS_SHARED = "minThreadsShared";
+    private static final String CONFIG_MAX_THREADS_SHARED = "maxThreadsShared";
+    private static final String CONFIG_KEEP_ALIVE_SHARED = "keepAliveTimeoutShared";
+    private static final String CONFIG_MIN_THREADS_CUSTOM = "minThreadsCustom";
+    private static final String CONFIG_MAX_THREADS_CUSTOM = "maxThreadsCustom";
+    private static final String CONFIG_KEEP_ALIVE_CUSTOM = "keepAliveTimeoutCustom";
+
+    private static final int MIN_CONSUMER_NAME_LENGTH = 4;
+    private static final int MAX_CONSUMER_NAME_LENGTH = 20;
+    private static final Pattern CONSUMER_NAME_PATTERN = Pattern.compile("[a-zA-Z0-9_\\-]*");
+
+    private volatile TrustManagerProvider trustmanagerProvider;
+
+    private ThreadPoolExecutor threadPool;
+    private HttpClient sharedHttpClient;
+    private HttpClientFacade sharedHttpClientFacade;
+
+    private int minThreadsShared;
+    private int maxThreadsShared;
+    private int keepAliveTimeoutShared;
+    private int minThreadsCustom;
+    private int maxThreadsCustom;
+    private int keepAliveTimeoutCustom;
+
+    @Activate
+    protected void activate(Map<String, Object> parameters) {
+        minThreadsShared = (int) parameters.get(CONFIG_MIN_THREADS_SHARED);
+        maxThreadsShared = (int) parameters.get(CONFIG_MAX_THREADS_SHARED);
+        keepAliveTimeoutShared = (int) parameters.get(CONFIG_KEEP_ALIVE_SHARED);
+        minThreadsCustom = (int) parameters.get(CONFIG_MIN_THREADS_CUSTOM);
+        maxThreadsCustom = (int) parameters.get(CONFIG_MAX_THREADS_CUSTOM);
+        keepAliveTimeoutCustom = (int) parameters.get(CONFIG_KEEP_ALIVE_CUSTOM);
+        // We need an "empty" queue, because otherwise jetty workers become queued instead of executed.
+        // This will cause jetty to hang!
+        // SynchronousQueue is fine for this, because it is effectively an empty queue
+        threadPool = createThreadPool("common", minThreadsShared, maxThreadsShared, keepAliveTimeoutShared);
+
+        sharedHttpClient = createHttpClientInternal("common", null);
+        sharedHttpClientFacade = new HttpClientDelegate(sharedHttpClient);
+
+        logger.info("jetty shared http client created");
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> parameters) {
+        minThreadsShared = (int) parameters.get(CONFIG_MIN_THREADS_SHARED);
+        maxThreadsShared = (int) parameters.get(CONFIG_MAX_THREADS_SHARED);
+        keepAliveTimeoutShared = (int) parameters.get(CONFIG_KEEP_ALIVE_SHARED);
+        minThreadsCustom = (int) parameters.get(CONFIG_MIN_THREADS_CUSTOM);
+        maxThreadsCustom = (int) parameters.get(CONFIG_MAX_THREADS_CUSTOM);
+        keepAliveTimeoutCustom = (int) parameters.get(CONFIG_KEEP_ALIVE_CUSTOM);
```

Comment:
Just an enhancement proposal: You might extract the properties extraction to a single method and use it in activate(..) and update(..) methods.

Reply:
You're right.